zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs

```diff
@@ -60,78 +49,45 @@ impl<F: Field> ExecutionGadget<F> for ErrorOOGCallGadget<F> {
             OpcodeId::CALL.expr(),
         );
 
-        let gas_word = cb.query_word();
-        let callee_address_word = cb.query_word();
-        let value = cb.query_word();
-        let cd_offset = cb.query_cell();
-        let cd_length = cb.query_rlc();
-        let rd_offset = cb.query_cell();
-        let rd_length = cb.query_rlc();
-        let balance = cb.query_word();
-
         let tx_id = cb.call_context(None, CallContextFieldTag::TxId);
-
         let is_static = cb.call_context(None, CallContextFieldTag::IsStatic);
-
-        // Lookup values from stack
-        cb.stack_pop(gas_word.expr());
-        cb.stack_pop(callee_address_word.expr());
-        cb.stack_pop(value.expr());
-        cb.stack_pop(cd_offset.expr());
-        cb.stack_pop(cd_length.expr());
-        cb.stack_pop(rd_offset.expr());
-        cb.stack_pop(rd_length.expr());
-        cb.stack_push(0.expr());
-
-        // Recomposition of random linear combination to integer
-        let callee_address =
-            from_bytes::expr(&callee_address_word.cells[..N_BYTES_ACCOUNT_ADDRESS]);
-        let cd_address = MemoryAddressGadget::construct(cb, cd_offset, cd_length);
-        let rd_address = MemoryAddressGadget::construct(cb, rd_offset, rd_length);
-        let memory_expansion =
-            MemoryExpansionGadget::construct(cb, [cd_address.address(), rd_address.address()]);
+        let call_gadget = CommonCallGadget::construct(cb, 1.expr(), 0.expr(), 0.expr());
 
         // Add callee to access list
         let is_warm = cb.query_bool();
-        cb.account_access_list_read(tx_id.expr(), callee_address.clone(), is_warm.expr());
+        cb.account_access_list_read(
+            tx_id.expr(),
+            call_gadget.callee_address_expr(),
+            is_warm.expr(),
+        );
 
-        let value_is_zero = IsZeroGadget::construct(cb, sum::expr(&value.cells));
-        let has_value = 1.expr() - value_is_zero.expr();
+        let balance = cb.query_word();
         cb.account_read(
-            callee_address.clone(),
+            call_gadget.callee_address_expr(),
             AccountFieldTag::Balance,
             balance.expr(),
         );
 
         // Verify gas cost
-        let [callee_nonce, callee_code_hash] = [AccountFieldTag::Nonce, AccountFieldTag::CodeHash]
-            .map(|field_tag| {
-                let value = cb.query_cell();
-                cb.account_read(callee_address.clone(), field_tag, value.expr());
-                value
-            });
+        let callee_nonce = cb.query_cell();
```

Comment:
CallOp already use `callee_exists` to calculate gas cost, here is also old way (check callee_nonce, balance empty to determine existence), would you like to update it in this PR?

Reply:
sure, I'll merge `main` first and request for your review again. Thanks!

Reply:
fixed in [e4f6656](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1023/commits/e4f6656be193e0b54b8c2da7b48a01d8ab7c9c4d)